### PR TITLE
Added comments

### PR DESCRIPTION
**Pull request by JacksonElia**
I just added some more comments. Once other PRs are merged I want to comment pyprojects.toml as well.

### CODE REVIEW

File: airbrakes/hardware/servo.py [hunk 1]
```diff
@@ -31,6 +40,7 @@ def set_extended(self) -> None:
         """
         Extends the servo to the maximum extension.
         """
+        # We have to use threading to avoid blocking the main thread because our extension methods sleep
```
Comment (harshil21) [inline airbrakes/hardware/servo.py:43]: huh I just realized there are absolutely no tests for this

File: airbrakes/hardware/servo.py [hunk 2]
```diff
@@ -10,12 +10,21 @@
 
 class Servo:
     """
-    A custom class that represents a servo motor, controlling the extension of the airbrakes.
+    A custom class that represents a servo motor, controlling the extension of the airbrakes. The servo is controlled
+    using the gpiozero library, which provides a simple interface for controlling GPIO pins on the Raspberry Pi.
```
Comment (harshil21) [inline airbrakes/hardware/servo.py:14]: don't you want to mention why we're using a bang bang controller and not PID?

File: airbrakes/mock/mock_imu.py [hunk 3]
```diff
@@ -49,10 +49,11 @@ def _fetch_data_loop(self, log_file_path: Path, real_time_simulation: bool) -> N
         Reads the data from the log file and puts it into the shared queue.
         :param log_file_path: the name of the log file to read data from located in logs/
         :param real_time_simulation: Whether to simulate a real flight by sleeping for a set period, or run at full
-            speed, e.g. for using it in the CI.
+        speed, e.g. for using it in the CI.
         """
         signal.signal(signal.SIGINT, signal.SIG_IGN)  # Ignore the KeyboardInterrupt signal
 
+        # Opens the log file
```
Comment (harshil21) [inline airbrakes/mock/mock_imu.py:56]: completely unnecessary comment imo

File: airbrakes/mock/mock_imu.py [hunk 4]
```diff
@@ -72,25 +73,30 @@ def _fetch_data_loop(self, log_file_path: Path, real_time_simulation: bool) -> N
                 est_linear_accel_x = row.get("estLinearAccelX")  # estimated data packet field
                 # Create the data packet based on the row
                 if scaled_accel_x:
+                    # If the row has the scaledAccelX field, then it's a raw data packet
```
Comment (harshil21) [inline airbrakes/mock/mock_imu.py:76]: already commented that above with the variable

File: airbrakes/mock/mock_imu.py [hunk 5]
```diff
@@ -72,25 +73,30 @@ def _fetch_data_loop(self, log_file_path: Path, real_time_simulation: bool) -> N
...
                     for key in RawDataPacket.__struct_fields__:
                         if val := row.get(key, None):
                             fields_dict[key] = convert_to_float(val)
                     fields_dict["timestamp"] = convert_to_nanoseconds(row["timestamp"])
                     imu_data_packet = RawDataPacket(**fields_dict)
                 elif est_linear_accel_x:
+                    # If the row has the estLinearAccelX field, then it's an estimated data packet
```
Comment (harshil21) [inline airbrakes/mock/mock_imu.py:83]: same

File: main.py [hunk 6]
```diff
@@ -26,14 +26,31 @@
 
 
 def main(is_simulation: bool, real_servo: bool) -> None:
+    """
+    The main function that will be run when the program is started. It will create the objects that will be used in the
+    airbrakes context and run the main loop. The main loop will run until the user presses Ctrl+C.
+
+    This function handles is what handles if the program is running in simulation mode or not.
```
Comment (harshil21) [inline main.py:33]: grammar

File: main.py [hunk 7]
```diff
@@ -26,14 +26,31 @@
...
     # Create the objects that will be used in the airbrakes context
+    sim_time_start = time.time()
     if is_simulation:
+        # If we are running a simulation, then we will replace our hardware objects with mock objects that just pretend
+        # to be the real hardware. This is useful for testing the software without having to fly the rocket.
+        # MockIMU pretends to be the imu by reading previous flight data from a log file
         imu = MockIMU(SIMULATION_LOG_PATH, real_time_simulation=True)
+        # MockFactory is used to create a mock servo object that pretends to be the real servo
         servo = Servo(SERVO_PIN) if real_servo else Servo(SERVO_PIN, pin_factory=MockFactory(pin_class=MockPWMPin))
+        print(f"\n{'='*10} REAL TIME FLIGHT DATA {'='*10}\n")
```
Comment (harshil21) [inline main.py:47]: ? revert 
```suggestion
```

File: main.py [hunk 8]
```diff
@@ -26,14 +26,31 @@
...
+    The main function that will be run when the program is started. It will create the objects that will be used in the
+    airbrakes context and run the main loop. The main loop will run until the user presses Ctrl+C.
+
+    This function handles is what handles if the program is running in simulation mode or not.
+
+    :param is_simulation: Whether to run the program in simulation mode or not
+    :param real_servo: Whether to use the real servo or a mock servo
+    """
     # Create the objects that will be used in the airbrakes context
+    sim_time_start = time.time()
```
Comment (harshil21) [inline main.py:39]: this will be done in the other PR, no need to duplicate

**Review by harshil21 — CHANGES_REQUESTED** [review]
some minor comments

File: airbrakes/hardware/servo.py (hunk 2 above)
Comment (JacksonElia) [reply airbrakes/hardware/servo.py:14]: the servo won't have the bang bang controller, that logic will most likely be in state.update()

File: airbrakes/mock/mock_imu.py (hunk 4 above)
Comment (JacksonElia) [reply airbrakes/mock/mock_imu.py:76]: woops i think that happened when i did the merge, i'll fix

File: main.py (hunk 8 above)
Comment (JacksonElia) [reply main.py:39]: wait so what do you want me to do

Comment (harshil21) [reply main.py:39]: delete that line in this pr, it already exists, and this is a redefinition

File: airbrakes/hardware/servo.py (hunk 2 above)
Comment (harshil21) [reply airbrakes/hardware/servo.py:14]: hm interesting

File: main.py [hunk 9]
```diff
@@ -26,14 +26,29 @@
 
 
 def main(is_simulation: bool, real_servo: bool) -> None:
+    """
+    The main function that will be run when the program is started. It will create the objects that will be used in the
+    airbrakes context and run the main loop. The main loop will run until the user presses Ctrl+C.
+
+    This function is what handles if the program is running in simulation mode or not.
```
Comment (harshil21) [inline main.py:33]: sorry, I still don't understand what you're trying to say. main() does not "handle" whether the program runs in sun mode or not. That's defined by the parameter real_time_sim

**Comment by JacksonElia** [comment]
Everything should be good? I resolved the merge conflicts, but it was really weird. It was like main already had some of my changes? Did we already merge another branch based off of this one into main?

**Review by harshil21 — APPROVED** [review]
> I resolved the merge conflicts, but it was really weird. It was like main already had some of my changes? Did we already merge another branch based off of this one into main?

Yeah, the way to resolve that would be by rebasing and force pushing. Anyway, the diff looks good and I will squash this commit so it doesn't really matter now.

**Pull request merged**